scripts/world_bank/worldbank.py

```diff
@@ -47,222 +47,347 @@
+
+    # Turn each year into its own row.
+    df = df.set_index(
+        ['CountryName', 'CountryCode', 'IndicatorName', 'IndicatorCode'])
+    df = df.stack()
+    df.index = df.index.rename('year', level=4)
+    df.name = "Value"
+    df = df.reset_index()
+
+    # Convert to numeric and drop empty values.
+    df['Value'] = pd.to_numeric(df['Value'])
+    df = df.dropna()
+    return df
 
-    Args:
-        country_iso: 3-digit ISO code for a country.
-
-    Notes:
-        Takes approximately 10 seconds to download and tidy one country in Google colab.
-  """
-
-  country_zip = f"http://api.worldbank.org/v2/en/country/{country_iso}?downloadformat=csv"
-  r = requests.get(country_zip)
-
-  filebytes = io.BytesIO(r.content)
-  myzipfile = zipfile.ZipFile(filebytes)  
-
-  # Payload file starts with API
-  file_to_open = None
-  for file in myzipfile.namelist():
-    if file.startswith("API"):
-      file_to_open = file
-      break
-
-  assert file_to_open != None
-
-  
-  df = None
-  line_match = re.compile(r"\"([^\"]*)\"")
-
-  for line in myzipfile.open(file_to_open).readlines():
-    # Cells are contained in quotations and comma separated
-    cols = line_match.findall(line.decode("utf-8"))
-    
-    # Exclude heading lines
-    if len(cols) > 2:
-      # Use first row as the header
-      if df is None:
-        df = pd.DataFrame(columns = cols)
-      else:
-        df = df.append(pd.DataFrame([cols], columns=df.columns), ignore_index=True)
-    
-  df = df.rename(columns=WORLDBANK_COL_REMAP)
-
-  # Turn each year into its own row
-  df = df.set_index(['CountryName', 'CountryCode', 'IndicatorName', 'IndicatorCode'])
-  df = df.stack()
-  df.index = df.index.rename('year', level=4)
-  df.name = "Value"
-  df = df.reset_index()
-
-  # Convert to numeric and drop empty values
-  df['Value'] = pd.to_numeric(df['Value'])
-  df = df.dropna()
-
-  return df
 
 def build_stat_vars_from_indicator_list(row):
-  """ Generates a WorldBank stat var for a row in the indicators dataframe """
-
-  def row_to_constraints(row):
-    """ Helper to generate list of constraints """
-    constraints_text = ""
-    next_constraint = 1
-
-    while next_constraint <= MAX_CONSTRAINTS and not pd.isna(row[f"p{next_constraint}"]):
-      variable = row[f'p{next_constraint}']
-      constraint = row[f'v{next_constraint}']
-      constraints_text += f"{variable}: {constraint}\n"
-      next_constraint += 1
+    """ Generates WorldBank StatVar for a row in the indicators dataframe. """
+
+    def row_to_constraints(row):
+        """ Helper to generate list of constraints. """
+        constraints_text = ""
+        next_constraint = 1
+        while (f"p{next_constraint}" in row and
+                not pd.isna(row[f"p{next_constraint}"])):
+            variable = row[f'p{next_constraint}']
+            constraint = row[f'v{next_constraint}']
+            constraints_text += f"{variable}: dcs:{constraint}\n"
+            next_constraint += 1
```

Comment:
So the processing should stop as soon as an NA is found, instead of just skipping over the NA?  If so, perhaps a comment indicating the row's structure would be helpful.  E.g., "'row' lists key-value pairs and ends with an NA."